gimme_aws_creds/webauthn.py

```diff
@@ -71,9 +72,12 @@ def on_keepalive(self, status):
 
     def work(self, client):
         try:
-            self._assertions, self._client_data = client.get_assertion(
-                self._rp['id'], self._challenge, self._allow_list, timeout=self._cancel, on_keepalive=self.on_keepalive
-            )
+            pin = None
+            if client.info.options.get("clientPin"):
+                # Prompt for PIN if needed
+                pin = getpass("Please enter PIN: ")
+            request_options=PublicKeyCredentialRequestOptions(challenge=base64.urlsafe_b64decode(self._challenge), rp_id=self._rp['id'], allow_credentials=self._allow_list)

```

Comment:
Better use `fido2.utils.websafe_decode` for decoding the challenge, as it might be truncated
from padding.

Reply:
Updated